applications/EigenSolversApplication/CMakeLists.txt

```diff
@@ -1,123 +1,140 @@
+        endif()
+    elseif( DEFINED ENV{CONDA_PREFIX} )
+        message("-- Found Conda environment: $ENV{CONDA_PREFIX}")
+
+        if( WIN32 )
+            include_directories(SYSTEM $ENV{CONDA_PREFIX}/Library/include)
+            link_directories("$ENV{CONDA_PREFIX}/Library/lib")
+        else()
+            include_directories(SYSTEM $ENV{CONDA_PREFIX}/include)
+            link_directories("$ENV{CONDA_PREFIX}/lib")
+        endif()
+    else()
+        message( FATAL_ERROR "MKLROOT not defined" )
+    endif()
+
+    add_definitions( -DUSE_EIGEN_MKL -DEIGEN_USE_MKL_ALL )
+endif()
+
+if( USE_EIGEN_FEAST MATCHES ON )
+    if( WIN32 )
+        message( FATAL_ERROR "FEAST4 is not yet supported under Windows")
+    endif()
+    if( USE_EIGEN_MKL MATCHES OFF OR NOT DEFINED USE_EIGEN_MKL )
+        message( FATAL_ERROR "FEAST4 can only be used with MKL" )
+    endif()
+    include(CMakeAddFortranSubdirectory)
+    cmake_add_fortran_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external_libraries/FEAST NO_EXTERNAL_INSTALL )
+    set(FEAST_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/external_libraries/FEAST/4.0 )
+    include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/external_libraries/FEAST/4.0/include )
+    add_definitions(-DUSE_EIGEN_FEAST)
+endif()
+
+## generate variables with the sources
+set( KRATOS_EIGENSOLVERS_APPLICATION_CORE_SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/eigen_solvers_application.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_factories/dense_linear_solver_factory.cpp
+)
+
+# Sources for the Python module
+file(
+    GLOB_RECURSE
+    KRATOS_EIGEN_PYTHON_INTERFACE_SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/*.cpp
+)
+
+###############################################################
+## EigenSolversApplication core library (C++ parts)
+add_library( KratosEigenSolversCore SHARED ${KRATOS_EIGENSOLVERS_APPLICATION_CORE_SOURCES} )
+target_link_libraries( KratosEigenSolversCore PUBLIC KratosCore )
+set_target_properties( KratosEigenSolversCore PROPERTIES COMPILE_DEFINITIONS "EIGENSOLVERS_APPLICATION=EXPORT,API")
+
+## EigenSolversApplication python module
+pybind11_add_module( KratosEigenSolversApplication MODULE THIN_LTO ${KRATOS_EIGEN_PYTHON_INTERFACE_SOURCES} )
+target_link_libraries( KratosEigenSolversApplication PRIVATE KratosEigenSolversCore )
+set_target_properties( KratosEigenSolversApplication PROPERTIES PREFIX "")
+
+if( USE_EIGEN_MKL MATCHES ON )
+    if( ${CMAKE_SYSTEM_NAME} MATCHES "Windows" )
+        find_library(MKL_RT_LIB mkl_rt)
+        message( "mkl_rt.lib found at: ${MKL_RT_LIB}")
+        target_link_libraries( KratosEigenSolversCore PUBLIC ${MKL_RT_LIB} )
+    elseif( ${CMAKE_CXX_COMPILER_ID} MATCHES Clang )
+        message( FATAL_ERROR "Clang does not yet support MKL" )
+    else( ${CMAKE_SYSTEM_NAME} MATCHES "Windows" )
+        target_link_libraries( KratosEigenSolversCore PUBLIC mkl_rt )
+        if( USE_EIGEN_FEAST MATCHES ON )
+            target_link_libraries(KratosEigenSolversCore PUBLIC feast4 gfortran m)

```

Comment:
I thought it is not yet supported in Win?

Reply:
it is not supported. This else belongs to `if( ${CMAKE_SYSTEM_NAME} MATCHES "Windows" )`. I find this cmake syntax quite irritating, but did not change it, as it was there before